src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -204,7 +204,7 @@ public void ExecuteInitializeMethod(HostSessionStartAnalysisScope sessionScope,
             // The Initialize method should be run asynchronously in case it is not well behaved, e.g. does not terminate.
             ExecuteAndCatchIfThrows(
                 sessionScope.Analyzer,
-                data => data.analyzer.Initialize(data.context),
+                static data => data.analyzer.Initialize(data.context),
```

Comment:
made all lambdas explicitly static to ensure we don't accidentally capture in them.